File: php/tests/Unit/AssessService/WordScorerTest.php
<?php

namespace FactChecker\Tests\Unit\AssessService;

use FactChecker\AssessService\WordScorer;
use PHPUnit\Framework\TestCase;

class WordScorerTest extends TestCase
{
    private WordScorer $scorer;

    protected function setUp(): void
    {
        $this->scorer = new WordScorer();
    }

    /** @test */
    public function it_can_score_a_sentence_without_cats()
    {
        $sentence = 'this is a test sentence without target';

        $this->assertSame(0, $this->scorer->calculateScore($sentence));
    }

    /** @test */
    public function it_can_score_a_sentence_with_one_cat()
    {
        $sentence = 'this is a test sentence with one cat';

        $this->assertSame(2, $this->scorer->calculateScore($sentence));
    }

    /** @test */
    public function it_can_score_a_sentence_with_two_cats()
    {
        $sentence = 'this is a test sentence with cat and cat';

        $this->assertSame(3, $this->scorer->calculateScore($sentence));
    }

    /** @test */
    public function it_can_score_a_sentence_with_three_acats()
    {
        $sentence = 'this is a test sentence with cats, cats, and cats';

        $this->assertSame(4, $this->scorer->calculateScore($sentence));
    }

    /** @test */
    public function it_can_score_a_sentence_with_three_and_more_cats()
    {
        $sentence = 'this is a test sentence with cats, cats, cats, and cats';

        $this->assertSame(4, $this->scorer->calculateScore($sentence));
    }
}

File: java/app/src/test/java/factchecker/AssessService/WordScorerTest.java
package factchecker.AssessService;

import static org.junit.jupiter.api.Assertions.assertSame;

import org.junit.jupiter.api.Test;

class WordScorerTest {
    private final WordScorer scorer = new WordScorer();

    @Test
    void it_can_score_a_sentence_without_cats() {
        String sentence = "this is a test sentence without target";

        assertSame(0, scorer.calculateScore(sentence));
    }

    @Test
    void it_can_score_a_sentence_with_one_cat() {
        String sentence = "this is a test sentence with one cat";

        assertSame(2, scorer.calculateScore(sentence));
    }

    @Test
    void it_can_score_a_sentence_with_two_cats() {
        String sentence = "this is a test sentence with cat and cat";

        assertSame(3, scorer.calculateScore(sentence));
    }

    @Test
    void it_can_score_a_sentence_with_three_cats() {
        String sentence = "this is a test sentence with cats, cats, and cats";

        assertSame(4, scorer.calculateScore(sentence));
    }

    @Test
    void it_can_score_a_sentence_with_three_and_more_cats() {
        String sentence = "this is a test sentence with cats, cats, cats, and cats";

        assertSame(4, scorer.calculateScore(sentence));
    }
}
